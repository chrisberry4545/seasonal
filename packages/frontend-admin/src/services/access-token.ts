import { getStorage } from '../consts';

export const setAccessToken = (token: string) => {
  getStorage().accessToken = token;
};

export const getAccessTokenHeaders = (): HeadersInit => {
  const token = getStorage().accessToken;
  return token
  ? {
    Authorization: `Bearer ${token}`
  }
  : {};
};
