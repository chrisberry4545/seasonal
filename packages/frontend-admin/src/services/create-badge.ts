import { IBadge } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { BADGE_URL } from '../config';

export const createBadge = async (badge: IBadge): Promise<IBadge> =>
  makeAuthorizedRequest<IBadge>(BADGE_URL, {
    body: JSON.stringify(badge),
    method: 'POST'
  });
