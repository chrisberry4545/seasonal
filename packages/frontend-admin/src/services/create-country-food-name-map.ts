import { ICountryFoodNameMap } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_FOOD_NAME_MAP_URL } from '../config';

export const createCountryFoodNameMap = async (
  country: ICountryFoodNameMap
): Promise<ICountryFoodNameMap> =>
  makeAuthorizedRequest<ICountryFoodNameMap>(COUNTRY_FOOD_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
