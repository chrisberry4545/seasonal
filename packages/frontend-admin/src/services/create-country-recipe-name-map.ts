import { ICountryRecipeNameMap } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_RECIPE_NAME_MAP_URL } from '../config';

export const createCountryRecipeNameMap = async (
  country: ICountryRecipeNameMap
): Promise<ICountryRecipeNameMap> =>
  makeAuthorizedRequest<ICountryRecipeNameMap>(COUNTRY_RECIPE_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
