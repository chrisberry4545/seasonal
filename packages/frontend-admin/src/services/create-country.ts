import { ICountry } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_URL } from '../config';

export const createCountry = async (country: ICountry): Promise<ICountry> =>
  makeAuthorizedRequest<ICountry>(COUNTRY_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
