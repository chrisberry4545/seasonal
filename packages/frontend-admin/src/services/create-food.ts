import { IFood } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { FOOD_URL } from '../config';

export const createFood = async (food: IFood): Promise<IFood> =>
  makeAuthorizedRequest<IFood>(FOOD_URL, {
    body: JSON.stringify(food),
    method: 'POST'
  });
