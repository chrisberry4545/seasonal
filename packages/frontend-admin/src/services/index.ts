export * from './access-token';

export * from './get-all-countries';

export * from './get-all-food';

export * from './get-all-recipes';

export * from './get-single-recipe';

export * from './login-request';

export * from './make-authorized-request';

export * from './update-recipe';
