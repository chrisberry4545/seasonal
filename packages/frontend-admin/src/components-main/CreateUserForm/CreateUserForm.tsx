import React, { FC } from 'react';
import { createUser } from '../../services';
import { IUser } from '@chrisb-dev/seasonal-shared-models';
import { BaseFormUser } from '../BaseFormUser/BaseFormUser';
import { FORM_BUTTON_TEXT } from '../../consts';
import { LayoutWithTitle } from '../../components-layouts';

const createEmptyUserItem = (): IUser => ({
  password: '',
  roles: [] as string[],
  username: ''
} as IUser);

export const CreateUserForm: FC<{}> = () =>
  <LayoutWithTitle title='Create User'>
    <BaseFormUser items={createEmptyUserItem()}
        updateMethod={createUser}
        buttonText={FORM_BUTTON_TEXT.CREATE} />
  </LayoutWithTitle>;
