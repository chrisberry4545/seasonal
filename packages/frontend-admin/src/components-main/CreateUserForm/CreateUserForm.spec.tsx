
import React from 'react';
import { CreateUserForm } from './CreateUserForm';
import { shallow, ShallowWrapper } from 'enzyme';
import { IUser } from '@chrisb-dev/seasonal-shared-models';
import * as services from '../../services';
import {
  BaseFormUser
} from '../BaseFormUser/BaseFormUser';

jest.mock('../../services', () => ({
  createUser: jest.fn()
}));
jest.mock('../../consts', () => ({
  FORM_BUTTON_TEXT: {
    CREATE: 'create'
  }
}));
jest.mock('../../components-layouts', () => ({
  LayoutWithTitle: () => 'LayoutWithTitle'
}));
jest.mock('../BaseFormUser/BaseFormUser', () => ({
  BaseFormUser: () => 'BaseFormUser'
}));

describe('<CreateUserForm />', () => {
  let wrapper: ShallowWrapper;
  let mockCreate: jest.SpyInstance;

  beforeEach(() => {
    mockCreate = jest.spyOn(services, 'createUser')
      .mockResolvedValue({} as IUser);
    wrapper = shallow(
      <CreateUserForm />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('calls create on update', () => {
    const item = {} as IUser;
    const update = wrapper.find(BaseFormUser).props().updateMethod;
    if (update) {
      update(item);
    }
    expect(mockCreate).toHaveBeenCalledWith(item);
  });

});
