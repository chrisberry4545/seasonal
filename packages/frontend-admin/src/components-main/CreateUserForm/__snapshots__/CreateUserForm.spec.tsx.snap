// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUserForm /> renders correctly 1`] = `
<LayoutWithTitle
  title="Create User"
>
  <BaseFormUser
    buttonText="create"
    items={
      Object {
        "password": "",
        "roles": Array [],
        "username": "",
      }
    }
    updateMethod={[MockFunction]}
  />
</LayoutWithTitle>
`;
